### PR TITLE
Discover tests using save-core

### LINKED ISSUE #133: Preprocessor: Support configs discovering using save-common
After #114 is done and proper APIs are create in save-common

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Changed logic
* Added tests
* Fixed link to HistoryView
* Reverted liquibase version
* Added a common save-cli snapshot version accessor for agent and orchestrator

Closes #133

### CODE REVIEW

File: save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/service/TestDiscoveringService.kt [hunk 1]
```diff
@@ -20,79 +20,66 @@ import org.springframework.stereotype.Service
...
+    /**
+     * Returns a root config of hierarchy; this config will already have all descendants merged with their parents.
+     *
+     * @param testResourcesRootAbsolutePath path to directory with root config
+     * @return a root [TestConfig]
+     */
+    @OptIn(ExperimentalFileSystem::class)
+    fun getRootTestConfig(testResourcesRootAbsolutePath: String): TestConfig =
+            ConfigDetector().configFromFile(testResourcesRootAbsolutePath.toPath()).apply {
+                getAllTestConfigs().onEach {
```
Comment (kentr0w) [inline save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/service/TestDiscoveringService.kt:33]: Is it possible, that there is no any config file?

**Review by kentr0w — CHANGES_REQUESTED** [review]

File: save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/service/TestDiscoveringService.kt (hunk 1 above)
Comment (petertrr) [reply save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/service/TestDiscoveringService.kt:33]: Then `ConfigDetector().configFromFile` will throw `IllegalArgumentException` and execution will be marked as failed in `onErrorResume`. I'll add a note about `IAE` in this method's KDoc

File: save-orchestrator/build.gradle.kts [hunk 2]
```diff
@@ -19,17 +19,15 @@ tasks.withType<KotlinCompile> {
...
-tasks.getByName("processResources").finalizedBy("downloadSaveCli")
+// if required, file can be provided manually
 val saveCliVersion = getSaveCliVersion()
-tasks.register<Download>("downloadSaveCli") {
-    dependsOn("processResources")
-    // if required, file can be provided manually
-    if (file("$buildDir/resources/main/save-$saveCliVersion-linuxX64.kexe").exists()) {
-        src(file("$buildDir/resources/main/save-$saveCliVersion-linuxX64.kexe").toURI().toURL())
-    } else {
+if (!file("$buildDir/resources/main/save-$saveCliVersion-linuxX64.kexe").exists()) {
```
Comment (kentr0w) [inline save-orchestrator/build.gradle.kts:24]: What if file exists?

**Review by kentr0w — CHANGES_REQUESTED** [review]

File: save-orchestrator/build.gradle.kts (hunk 2 above)
Comment (petertrr) [reply save-orchestrator/build.gradle.kts:24]: Then it will be included in the image, no need to download anything

**Review by kentr0w — APPROVED** [review]
lgtm

**Pull request merged**